plugins/utils/auto-save.nix

```diff
@@ -6,65 +6,94 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.auto-save;
-in
-{
-  options.plugins.auto-save = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "auto-save";
-
-    package = helpers.mkPluginPackageOption "auto-save" pkgs.vimPlugins.auto-save-nvim;
-
-    keymaps = {
-      silent = mkOption {
-        type = types.bool;
-        description = "Whether auto-save keymaps should be silent.";
-        default = false;
-      };
-
-      toggle = helpers.mkNullOrOption types.str "Keymap for running auto-save.";
-    };
```

Comment:
The keymaps option can be maintained by putting it in `extraOptions` (as opposed to `settingsOptions`)